File: one_million/optimized/one_million.rb
#!/usr/bin/env ruby

$stdout.sync = false
prefix = "Hello, this is iteration number: ".freeze
chunk = String.new(capacity: 65536)

1_000_000.times do |n|
  chunk << prefix << n.to_s << "\n"
  if chunk.bytesize >= 60000
    $stdout.write(chunk)
    chunk.clear
  end
end

$stdout.write(chunk) unless chunk.empty?

File: one_million/optimized/one_million.js
const prefix = "Hello, this is iteration number: ";
const CHUNK_SIZE = 10000;
let chunk = "";

for (let i = 0; i < 1000000; i++) {
	chunk += prefix + i + "\n";
	if (i % CHUNK_SIZE === CHUNK_SIZE - 1) {
		process.stdout.write(chunk);
		chunk = "";
	}
}
if (chunk.length > 0) {
	process.stdout.write(chunk);
}
